src/azure-cli-core/azure/cli/core/keys.py

```diff
@@ -32,14 +35,33 @@ def is_valid_ssh_rsa_public_key(openssh_pubkey):
 def generate_ssh_keys(private_key_filepath, public_key_filepath):
     import paramiko
 
+    if os.path.exists(public_key_filepath):
+        with open(public_key_filepath, 'r') as public_key_file:
+            public_key = public_key_file.read()
+
+            pub_ssh_dir, _ = os.path.split(public_key_filepath)
+            logger.warning("Public SSH key file '%s' found in dir '%s'."
+                           "Use public key file. New RSA key pair will not be generated.",
+                           public_key_filepath, pub_ssh_dir)
+
+            return public_key
+
     ssh_dir, _ = os.path.split(private_key_filepath)
     if not os.path.exists(ssh_dir):
         os.makedirs(ssh_dir)
         os.chmod(ssh_dir, 0o700)
 
-    key = paramiko.RSAKey.generate(2048)
-    key.write_private_key_file(private_key_filepath)
-    os.chmod(private_key_filepath, 0o600)
+    try:
+        # try to use existing private key if it exists.
+        key = paramiko.RSAKey(filename=private_key_filepath)
+        logger.warning("Private SSH key file '%s' found in dir '%s'."
+                       " Generating public key file '%s'",
+                       private_key_filepath, ssh_dir, public_key_filepath)
+    except IOError:
```

Comment:
You only want to catch a FileNotFoundError right?
It may be better to do the existence check like you do with the public key.

Reply:
Okay, I will change this to check the path instead